tests/Functional/ORMModelFactoryTest.php

```diff
@@ -493,6 +495,24 @@ public function embeddables_are_never_persisted(): void
         $this->assertSame('another address', $object2->getValue());
     }
 
+    /**
+     * @test
+     */
+    public function can_create_factories_for_rich_domain_models(): void
+    {
+        $post = RichPostFactory::createOne([
+            'comments' => RichCommentFactory::new()->many(5),
+        ]);
+
+        RichPostFactory::assert()->count(1);
+        RichCommentFactory::assert()->count(5);
+        $this->assertCount(5, $post->getComments());
+
+        foreach (RichCommentFactory::all() as $comment) {
+            $this->assertSame($post->getId(), $comment->getPost()->getId());
```

Comment:
What about asserting between objects instead of scalars?
```suggestion
            $this->assertSame($post, $comment->getPost());
```